compiler/crates/relay-transforms/src/validations/validate_module_names/extract_module_name.rs

```diff
@@ -30,7 +30,7 @@ pub fn extract_module_name(path: &str) -> Option<String> {
 
 fn get_final_non_index_js_segment(path: &Path) -> Option<&str> {
     let file_stem = path.file_stem()?.to_str();
-    if file_stem == Some("index") {
+    if file_stem.map_or(false, |f| f.starts_with("index")) {
```

Comment:
should this be `starts_with("index.")`? otherwise this would return the parent for a file like `indexFoo.js` which i don't think is intended

Reply:
good catch, thanks